src/audio/audio_stream.c

```diff
@@ -141,6 +141,7 @@ void audio_stream_init(struct audio_stream *audio_stream, void *buff_addr, uint3
 	audio_stream->addr = buff_addr;
 	audio_stream->end_addr = (char *)audio_stream->addr + size;
 
+	audio_stream_init_alignment_constants(1, 1, audio_stream);
```

Comment:
the essence is to set: align_frame_cnt and align_shift_idx, do you think current function name exactly reflect the content?
I mean good to change function name, due to 1, 1 params make people don't know exactly what it is setting.

Reply:
actually, at this point, I'am afraid the format/channel of audio_stream haven't been set yet, wouldn't the set in audio_stream_set_params enough?

Reply:
@andyross @andrula-song I wonder if we should start by just applying the first patch. Setting the alignment constraints before channel count and sample format is not correct, but at least we'd get a determistic behaviour in case some module does not correctly set the constraints in its prepare functions. Or alternatively initialize the values to some known invalid value to ensure we catch these cases. 

We discussed earlier that we could have update hooks to recalculate the alignment constants whenever channel count or sample format is updated, but this is not so straightforward currently as the raw alignment input values (the constraints coming from algorithm) are not directly stored in audio stream.

Currently, the modules do need to call this in their prepare.

Reply:
>  I wonder if we should start by just applying the first patch.

The second patch is a behavioral noop, it only removes the code that was recapitulating the defaults.  Though it would be sad to lose the cleanup.  I feel really strongly at this point that we want **all** of this "just override the topology/framework values to make it work" code out of the components entirely.  Buffers should be write-only from component code.

Reply:
@andyross Hmm, but the audio_stream_init_alignment_constants() is really a convenience/helper interface between the audio-stream (and now sink/source) and the module. For this helper interface to function correctly, algorithm must use it once it knows the channel count and sample format. The "byte_align==1" is indeed odd, but it's ignored if sample size is larger (and it always is). So in practise these calls set the alignment based on the stream's channel count and sample size. 

Maybe this is correct, but it's not obvious to me the result will be the same alignment in all cases with combination of these two patches. I'd like +1 from @singalsu @andrula-song et al this is valid. The first patch alone seems safe, as it sets the defaults to sane values.

Reply:
Oh... OK, now I'm understanding you.  To restate to make sure I have it:  init_alignment_constants() takes as input the frame size from the stream struct, but at creation time that doesn't exist.  So in general it produces different values in when called in this patch vs. at runtime in the components.  But that maybe that's OK, since the components are only doing this to express "don't care" alignment requirements anyway and any value that doesn't break should work.

So I actually agree, we should definitely remove the second patch if so, and replace it with e.g. code in ipc_comp_connect() (both versions) to do this on behalf of default components.

Reply:
@kv2019i @andyross , if you indeed want to apply an incorrect value first and set the specific value for each component, I'am okay, but I recall some one who report an issue by setting the meaningless value while the format and channel not known here.(sorry, can not find that issue id).